server/src/main/java/org/opensearch/index/ReplicationStats.java

```diff
@@ -66,11 +81,20 @@ public long getMaxReplicationLag() {
         return this.maxReplicationLag;
     }
 
+    public long getTotalRejections() {
+        return totalRejections;
+    }
+
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         out.writeVLong(maxBytesBehind);
         out.writeVLong(totalBytesBehind);
         out.writeVLong(maxReplicationLag);
+        // TODO: change to V_2_11_0 on main after backport to 2.x
+        if (out.getVersion().onOrAfter(Version.CURRENT)) {
+            out.writeVLong(totalRejections);
+            out.writeOptionalWriteable(shardId);
```

Comment:
SegRep backpressure rejections are tracked at the replication group level. To avoid double-counting these rejections due to replicas + primary during rollup, I am checking if the shardId is same. If it is same, we don't sum the two rejection counts, else we do.

Reply:
Resolving this thread as the PR now adds only remote store stats